### DISCUSSION TITLE
Plasma frequency time step condition

### DISCUSSION THREAD

**Discussion by michelosca** [epochpic/epoch#140] [General]
I think that, the subroutine "set_plasma_frequency_dt" in epoch1d/src/housekeeping that sets the minimum time step required by the plasma, uses an erroneous maximum wavenumber value (k_max). Currently it is defined as k_max = 2.0_num * pi / dx but the maximum possible wavelength in a simulation domain with cell width dx is 2.0*dx hence k_max = pi / dx (the 2.0 factor may be not necessary).

I have noticed this in the 1D case, I do not know whether this is also the case in 2 and 3D.

**Comment by TomGoffrey** [comment]
Assuming you mean the minimum wavelength, I agree.

I am hesitant to change this without also modifying the code to recalculate omega during the calculation. As it stands (only calculating omega at t=0), I think I'd vote for keeping the current (overly restrictive) calculation.
